Firmware/tmc2130.cpp

```diff
@@ -379,56 +393,84 @@ void tmc2130_check_overtemp()
 
 void tmc2130_setup_chopper(uint8_t axis, uint8_t mres, uint8_t current_h, uint8_t current_r)
 {
-	uint8_t intpol = (mres != 0); // intpol to 256 only if microsteps aren't 256
+	union ChopConfU {
+		struct __attribute__((packed)) S {
+			uint32_t toff : 4;
+			uint32_t hstrt : 3;
+			uint32_t hend : 4;
+			uint32_t fd : 1;
+			uint32_t disfdcc : 1;
+			uint32_t rndtf : 1;
+			uint32_t chm : 1;
+			uint32_t tbl : 2;
+			uint32_t vsense : 1;
+			uint32_t vhighfs : 1;
+			uint32_t vhighchm : 1;
+			uint32_t sync : 4;
+			uint32_t mres : 4;
+			uint32_t intpol : 1;
+			uint32_t dedge : 1;
+			uint32_t diss2g : 1;
+			uint32_t reserved : 1;
+		} s;
+		uint32_t dw;
```

Comment:
I'd recommend adding a constexpr constructor to set all the members. It might look cumbersome, but the compiler should optimize most of the initialization code below better.

Reply:
Done